src/extension/netcdf/src/main/java/org/geoserver/netcdf/NetCDFUnitFormatInstance.java

```diff
@@ -0,0 +1,18 @@
+/* (c) 2018 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+
+package org.geoserver.netcdf;
+
+import org.geotools.imageio.netcdf.NetCDFUnitFormat;
```

Comment:
This is not going to work. The units configuration is there to support reads too (see VariableAdapter usage in GeoTools), otherwise the class would be placed in netcdf-out.

However, I don't see why you can't use NetCDFUnitFormat directly, and go for a different instance instead.
If mutability is the issue, thinking out loud, we could have a way to build a new NetCDFUnitFormat internal instance, and just throw away the old one: this way the unit format would be immutable, for as long as it lives. Just inform the callers not to cache the formatter instance, but always get it fresh from NetCDFUnitFormat.

Reply:
Thanks! You mean that we also use that instance from `VariableAdapter` then?

Reply:
I mean the configuration file in GeoServer should affect also VariableAdapter parsing. In other words, the configuration is used also to parse NetCDF files with uncommon units, and not only to control the generation of new NetCDFs in the GeoServer output format.